sktime/annotation/datagen.py

```diff
@@ -196,6 +196,77 @@ def piecewise_normal(
     return np.concatenate(tuple(segments_data))
 
 
+def piecewise_multinomial(
+    n_trials: int,
+    lengths: npt.ArrayLike,
+    pvals: npt.ArrayLike,
+    random_state: Union[int, np.random.RandomState] = None,
+) -> npt.ArrayLike:
+    """
+    Generate series from segments.
+
+    Each segment has length specified in ``lengths`` and data sampled from a multinomial
+    distribution with a total number of experiments for each trial set from ``n_trials``
+    and the probability for each outcome stored inside a list contained in  ``p_vals``.
+
+    Parameters
+    ----------
+    n_trials : int
+        Number of experiments to run during each trial
+    lengths : array_like
+        Lengths of the segments to be generated
+    pvals : array_like
+        Set of probabilities for each outcome for each distribution
+    random_state : int or np.random.RandomState
+        Either a random seed or RandomState instance
+
+    Returns
+    -------
+    data : np.array
+        univariate or multivariate time series as np.array
```

Comment:
It could be nice (from the user perspective) if you added a detail here about what is the expected size of this array and how that corresponds to the input variables (eg `lengths` and `n_trials`).